doc/source/ray-air/examples/serving_guide.ipynb

```diff
@@ -0,0 +1,658 @@
+    "\n",
+    "You can learn more about the http adapter [our user guide](serve-http-adapters)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\u001b[2m\u001b[36m(ServeController pid=66733)\u001b[0m INFO 2022-05-20 11:31:09,317 controller 66733 deployment_state.py:1217 - Adding 1 replicas to deployment 'DataFramePredictor'.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from ray.serve.http_adapters import pandas_read_json\n",
+    "\n",
+    "ModelWrapperDeployment.options(name=\"DataFramePredictor\").deploy(\n",
+    "    predictor_cls=DataFramePredictor,\n",
+    "    checkpoint=local_checkpoint,\n",
+    "    http_adapter=pandas_read_json\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's send a requests to our endpoint. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'[{\"base\":1,\"multipiler\":2,\"prediction\":4},{\"base\":3,\"multipiler\":4,\"prediction\":14}]'"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\u001b[2m\u001b[36m(HTTPProxyActor pid=66737)\u001b[0m INFO 2022-05-20 11:31:15,388 http_proxy 127.0.0.1 http_proxy.py:320 - POST /DataFramePredictor 200 28.6ms\n",
+      "\u001b[2m\u001b[36m(DataFramePredictor pid=66765)\u001b[0m INFO 2022-05-20 11:31:15,387 DataFramePredictor DataFramePredictor#lcGDjS replica.py:483 - HANDLE __call__ OK 24.4ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "resp = requests.post(\n",
+    "    \"http://localhost:8000/DataFramePredictor/\",\n",
+    "    json=[{\"base\": 1, \"multipiler\": 2}, {\"base\": 3, \"multipiler\": 4}],\n",
+    "    params={\"orient\": \"records\"},\n",
+    ")\n",
+    "resp.raise_for_status()\n",
+    "resp.text"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Great! You can see that the input JSON has been converted to dataframe and our predictor just need to worry about dataframe instead of the raw requests.\n",
+    "\n",
+    "But what if we need do to configure the HTTP request? You can do that as well.\n",
+    "\n",
+    "## 3. Accepting custom inputs via `http_adapter`\n",
+    "\n",
+    "The `http_adapter` field accept any callable function that's type annotated. You can also bring in additional types that's accepted by FastAPI's dependency injection framework. You can learn more detail [here](serve-http-adapters). In the following example, instead of using the pandas adapter Serve provides, we will implement our own request adapter that work with just http parameters instead of JSON."
```

Comment:
```suggestion
    "The `http_adapter` field accept any callable function that's type annotated. You can also bring in additional types that's accepted by FastAPI's dependency injection framework. To learn more, see [Serve HTTP Adapters](serve-http-adapters). In the following example, instead of using the pandas adapter Serve provides, we will implement our own request adapter that work with just http parameters instead of JSON."
```

https://developers.google.com/style/link-text